Apps/frontend/src/views/Oscilloscope.svelte

```diff
@@ -15,73 +15,100 @@
 
   let isEnabled = false;
 
+  // ----- Svelte lifecycle hooks -----
   onMount(() => {
-    socket = new WebSocket(import.meta.env.VITE_BACKEND_WEBSOCKET);
+    socket = createSocket();
+
+    socket.onopen = socketOnOpen;
+    socket.onmessage = socketOnMessage;
+    socket.onclose = socketOnClose;
+  });
+
+  onDestroy(() => {
```

Comment:
socket.onclosed is only called after socket.close() so if .close() is put inside onclose, it will never be called.
You can regard .onclose as a clean up function after the socket is closed

Reply:
Okay, my misunderstanding, thanks for your explanation.